modules/build/src/test/scala/scala/build/tests/SourcesTests.scala

```diff
@@ -501,4 +506,132 @@ class SourcesTests extends munit.FunSuite {
     }
   }
 
+  test("throw error when exclude found in multiple file") {
+    val testInputs = TestInputs(
+      os.rel / "Hello.scala" ->
+        """//> using exclude "*.sc"
+          |""".stripMargin,
+      os.rel / "Main.scala" ->
+        """//> using exclude "*/test/*"
+          |""".stripMargin
+    )
+    testInputs.withInputs { (_, inputs) =>
+      val crossSources =
+        CrossSources.forInputs(
+          inputs,
+          preprocessors,
+          TestLogger(),
+          SuppressWarningOptions()
+        )
+      crossSources match {
+        case Left(_: MultipleExcludeError) =>
+        case o                             => fail("Exception expected", clues(o))
+      }
+    }
+  }
+
+  test("exclude relative paths") {
+    val testInputs = TestInputs(
+      os.rel / "Hello.scala" -> "object Hello",
+      os.rel / "Main.scala" ->
+        """object Main {
+          |}""".stripMargin,
+      os.rel / "project.scala" ->
+        s"""//> using exclude "Main.scala" """
+    )
+    testInputs.withInputs { (_, inputs) =>
+      val (crossSources, _) =
+        CrossSources.forInputs(
+          inputs,
+          preprocessors,
+          TestLogger(),
+          SuppressWarningOptions()
+        ).orThrow
+      val scopedSources = crossSources.scopedSources(BuildOptions()).orThrow
+      val sources = scopedSources.sources(Scope.Main, crossSources.sharedOptions(BuildOptions()))
+
+      expect(sources.paths.nonEmpty)
+      expect(sources.paths.length == 2)
+      expect(sources.paths.map(_._2) == Seq(os.rel / "Hello.scala", os.rel / "project.scala"))
+    }
+  }
+
+  test("exclude absolute file paths") {
+    val testInputs = TestInputs(
+      os.rel / "Hello.scala" -> "object Hello",
+      os.rel / "Main.scala" ->
+        """object Main {
+          |}""".stripMargin,
+      os.rel / "project.scala" ->
+        s"""//> using exclude "$${.}${File.separator}Main.scala" """
```

Comment:
hmm, is that an absolute path? TBH I do not understand how this path work

Reply:
It was added a new special syntax `${.}` in using directive for handling path. More info you can find [here](https://github.com/VirtusLab/scala-cli/blob/main/website/docs/guides/using-directives.md#explicit-handling-of-paths-in-using-directives).